metricbeat/module/consul/agent/agent_integration_test.go

```diff
@@ -39,11 +42,27 @@ func TestFetch(t *testing.T) {
 	compose.EnsureUp(t, "consul")
 
 	f := mbtest.NewReportingMetricSetV2(t, consul.GetConfig([]string{"agent"}))
-	events, errs := mbtest.ReportingFetchV2(f)
-	if len(errs) > 0 {
-		t.Fatalf("Expected 0 error, had %d. %v\n", len(errs), errs)
+	var events []mb.Event
+
+	attempts := 0
+	var errs []error
+	for i := 0; i < attempts; i++ {
+		events, errs = mbtest.ReportingFetchV2(f)
+		if len(errs) > 0 {
+			t.Fatalf("Expected 0 errors, had %d. %v\n", len(errs), errs)
+		}
+
+		if len(events) == 0 {
+			attempts++
+			if attempts == 10 {
+				t.Fatalf("Expected 1 event, had %d\n", len(events))
+			}
+
+			time.Sleep(5 * time.Second)
```

Comment:
In the worst case this extends the full test suite by 50s for just one test. I would prefer having something in the healthcheck instead to makes sure the container is only reported as up, when it should be. 

What is the condition we are looking for to have at least 1 event?

Reply:
Yes. It was the fast approach without the PR skipping the test. Now that it's skipped I can try something else.

The idea is to emit one event, yes. But I just couldn't reproduce the error locally. It ALWAYS return events in less than a couple of seconds. My next idea was to write a local script with a for loop that launches the container, test and takes it down infinitely and reports every time that it fails. Yesterday I maybe try like 30 times and all of them were ok

Reply:
It seems this only happens in a very short period of time. So 50s sounds excessive. Perhaps we can make the wait time 200ms and have 50 attempts? So the max delay is 10s?

Reply:
Instead of sleeping, uses channels that will block until all the events are received. If the events are not received in a good amount of time the test will fail from the default timeout on the test suite.